Fixed #35591 -- Added warning to runserver about being unsuitable for production. (#18435)
django/django: docs/ref/django-admin.txt

This server uses the WSGI application object specified by the
:setting:`WSGI_APPLICATION` setting.

> .. warning::
> 
>     DO NOT USE THIS SERVER IN A PRODUCTION SETTING. It has not gone through
>     security audits or performance tests. (And that's how it's gonna stay. We're in
>     the business of making web frameworks, not web servers, so improving this
>     server to be able to handle a production environment is outside the scope of
>     Django.)

Reviewer: ```suggestion
.. warning:: DO NOT USE THIS SERVER IN A PRODUCTION SETTING.

    This server has not gone through security audits or performance tests, and
    hence is unsuitable for production. We're in the business of making web
    frameworks, not web servers, so improving this server to be able to handle a
    production environment is outside the scope of Django.
```

Uses IPv6 for the development server. This changes the default IP address from
``127.0.0.1`` to ``::1``.

> .. django-admin-option:: HIDE_PRODUCTION_WARNING
> 
>     .. versionadded:: 5.2

Reviewer: ```suggestion
.. envvar:: HIDE_PRODUCTION_WARNING

.. versionadded:: 5.2
```

> Setting this environment variable to the value of ``'true'`` will hides a warning
> about ``runserver`` not being suitable for production and a link to
> the deployment documentation.

Reviewer: ```suggestion
By default, a warning is printed to the console that ``runserver`` is not
suitable for production:

.. code-block:: text

    WARNING: This is a development server. Do not use it in a production setting. Use a production WSGI server instead.
    For more information on production servers see: https://docs.djangoproject.com/en/stable/howto/deployment/

Set this environment variable to ``"true"`` to hide this warning.
```
